### PR TITLE
Add style cop to more bindings

### PR DESCRIPTION
**Pull request by torbacz**
## Description

- Skip tests project in StyleCop syn scripts.
- Add style cop to A4988 project.
- Add style cop to Bno055 project.
- Add style cop to Button project.
- Add style cop to Bmp180 project.
- Add style cop to Buzzer project.
- Add style cop to Bmm150 project.
- Commented TestCompensateVector3 in Bmm150 project (not compiling and causing error after applying style cop)
- Add style cop to Bh1745 project.

## Motivation and Context

- Addresses nanoFramework/Home#1101

## How Has This Been Tested?
Local and pipeline build.

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [ ] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (add new Unit Test(s) or improved existing one(s), has no impact on code or features)
- [x] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project (only if there are changes in source code).
- [ ] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [x] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/master/CONTRIBUTING.md) document.
- [x] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).
- [ ] I have added new tests to cover my changes.

### CODE REVIEW

**Review by Ellerbach — APPROVED** [review]
Thanks a lot one more time! Yes, we can skip samples and tests for StyleCop, doesn't make sense. Seems like a conflict because of the version bump. Approving as always!

**Comment by torbacz** [comment]
@Ellerbach Conflicts resolved. Can be merged.

File: devices/Bmm150/Bmm150.cs [hunk 1]
```diff
@@ -13,79 +13,78 @@
...
         /// <summary>
-        /// Gets or sets Magnetometer calibration compensation vector
+        /// Gets or sets Magnetometer calibration compensation vector.
         /// </summary>
         public Vector3 CalibrationCompensation { get; set; } = new Vector3();
 
         /// <summary>
         /// Primary I2C address for the Bmm150
-        /// In the official sheet (P36) states that address is 0x13: https://github.com/m5stack/M5_BMM150/blob/master/src/M5_BMM150_DEFS.h#L163
+        /// In the official sheet (P36) states that address is 0x13: https://github.com/m5stack/M5_BMM150/blob/master/src/M5_BMM150_DEFS.h#L16.3.
```
Comment (josesimoes) [inline devices/Bmm150/Bmm150.cs:53]: This "fix" looks suspicious...

File: devices/Bmm150/Bmm150.cs [hunk 2]
```diff
@@ -13,79 +13,78 @@
...
-        /// <param name="shouldDispose">True to dispose the I2C device when class is disposed</param>
-        public Bmm150(I2cDevice i2cDevice, Bmm150I2cBase Bmm150Interface, bool shouldDispose = true)
+        /// <param name="i2cDevice">The I2C device.</param>
+        /// <param name="bmm150Interface">The specific interface to communicate with the Bmm150.</param>
+        /// <param name="shouldDispose">True to dispose the I2C device when class is disposed.</param>
+        public Bmm150(I2cDevice i2cDevice, Bmm150I2cBase bmm150Interface, bool shouldDispose = true)
         {
             _i2cDevice = i2cDevice ?? throw new ArgumentNullException(nameof(i2cDevice));
-            _Bmm150Interface = Bmm150Interface;
+            this.bmm150Interface = bmm150Interface;
```
Comment (josesimoes) [inline devices/Bmm150/Bmm150.cs:87]: We're avoiding `this` everywhere. Shouldn't this be `_bmm150Interface` instead?

File: devices/Bmm150/Bmm150.cs [hunk 3]
```diff
@@ -194,30 +193,30 @@ public void CalibrateMagnetometer(int numberOfMeasurements = 100)
...
+        /// <returns>Returns true if the version match.</returns>
         public bool IsVersionCorrect => ReadByte(Register.WIA) == 0x32;
 
         /// <summary>
-        /// Read the magnetometer without Bias correction and can wait for new data to be present
+        /// Read the magnetometer without Bias correction and can wait for new data to be present.
         /// </summary>
-        /// <param name="waitForData">true to wait for new data</param>
-        /// <returns>The data from the magnetometer</returns>
+        /// <param name="waitForData">True to wait for new data.</param>
```
Comment (josesimoes) [inline devices/Bmm150/Bmm150.cs:209]:
```suggestion
        /// <param name="waitForData"><c>true</c> to wait for new data.</param>
```

File: devices/Bmm150/Bmm150.cs [hunk 4]
```diff
@@ -194,30 +193,30 @@ public void CalibrateMagnetometer(int numberOfMeasurements = 100)
...
 
         /// <summary>
         /// Read the magnetometer without Bias correction and can wait for new data to be present
-        /// More info, permalink: https://github.com/BoschSensortec/BMM150-Sensor-API/blob/a20641f216057f0c54de115fe81b57368e119c01/bmm150.c#L921
+        /// More info, permalink: https://github.com/BoschSensortec/BMM150-Sensor-API/blob/a20641f216057f0c54de115fe81b57368e119c01/bmm150.c#L921.
         /// </summary>
-        /// <param name="waitForData">true to wait for new data</param>
-        /// <param name="timeout">timeout for waiting the data, ignored if waitForData is false</param>
-        /// <returns>The data from the magnetometer</returns>
+        /// <param name="waitForData">True to wait for new data.</param>
```
Comment (josesimoes) [inline devices/Bmm150/Bmm150.cs:217]:
```suggestion
        /// <param name="waitForData"><c>true</c> to wait for new data.</param>
```

File: devices/Bmm150/Bmm150.cs [hunk 5]
```diff
@@ -269,19 +268,19 @@ public Vector3 ReadMagnetometerWithoutCorrection(bool waitForData, TimeSpan time
...
         public Vector3 ReadMagnetometer(bool waitForData = true) => ReadMagnetometer(waitForData, DefaultTimeout);
 
         /// <summary>
-        /// Read the magnetometer with compensation calculation and can wait for new data to be present
+        /// Read the magnetometer with compensation calculation and can wait for new data to be present.
         /// </summary>
-        /// <param name="waitForData">true to wait for new data</param>
-        /// <param name="timeout">timeout for waiting the data, ignored if waitForData is false</param>
-        /// <returns>The data from the magnetometer</returns>
+        /// <param name="waitForData">True to wait for new data.</param>
```
Comment (josesimoes) [inline devices/Bmm150/Bmm150.cs:281]:
```suggestion
        /// <param name="waitForData"><c>true</c> to wait for new data.</param>
```

File: devices/Bmm150/Bmm150.cs [hunk 6]
```diff
@@ -194,30 +193,30 @@ public void CalibrateMagnetometer(int numberOfMeasurements = 100)
...
         /// <summary>
         /// Read the magnetometer without Bias correction and can wait for new data to be present
-        /// More info, permalink: https://github.com/BoschSensortec/BMM150-Sensor-API/blob/a20641f216057f0c54de115fe81b57368e119c01/bmm150.c#L921
+        /// More info, permalink: https://github.com/BoschSensortec/BMM150-Sensor-API/blob/a20641f216057f0c54de115fe81b57368e119c01/bmm150.c#L921.
         /// </summary>
-        /// <param name="waitForData">true to wait for new data</param>
-        /// <param name="timeout">timeout for waiting the data, ignored if waitForData is false</param>
-        /// <returns>The data from the magnetometer</returns>
+        /// <param name="waitForData">True to wait for new data.</param>
+        /// <param name="timeout">Timeout for waiting the data, ignored if waitForData is false.</param>
```
Comment (josesimoes) [inline devices/Bmm150/Bmm150.cs:218]:
```suggestion
        /// <param name="timeout">Timeout for waiting the data, ignored if <paramref name="waitForData"/> is <c>false<c/>.</param>
```

File: devices/Bmm150/Bmm150.cs [hunk 7]
```diff
@@ -269,19 +268,19 @@ public Vector3 ReadMagnetometerWithoutCorrection(bool waitForData, TimeSpan time
...
 
         /// <summary>
-        /// Read the magnetometer with compensation calculation and can wait for new data to be present
+        /// Read the magnetometer with compensation calculation and can wait for new data to be present.
         /// </summary>
-        /// <param name="waitForData">true to wait for new data</param>
-        /// <param name="timeout">timeout for waiting the data, ignored if waitForData is false</param>
-        /// <returns>The data from the magnetometer</returns>
+        /// <param name="waitForData">True to wait for new data.</param>
+        /// <param name="timeout">Timeout for waiting the data, ignored if waitForData is false.</param>
```
Comment (josesimoes) [inline devices/Bmm150/Bmm150.cs:282]:
```suggestion
        /// <param name="timeout">Timeout for waiting the data, ignored if <paramref name="waitForData"/> is <c>false<c/>.</param>
```

File: devices/Bmm150/Bmm150Compensation.cs [hunk 8]
```diff
@@ -29,19 +29,19 @@ public static double CompensateX(double x, uint rhall, Bmm150TrimRegisterData tr
             float processCompX2;
             float processCompX3;
             float processCompX4;
-            int BMM150_OVERFLOW_ADCVAL_XYAXES_FLIP = -4096;
+            int bmm150Overflow_adcval_xyaxes_flip = -4096;
```
Comment (josesimoes) [inline devices/Bmm150/Bmm150Compensation.cs:32]: Nit pick but we should be using camel case style on these, no need to introduce `_`

File: devices/Bmm150/Bmm150Compensation.cs [hunk 9]
```diff
@@ -68,19 +68,19 @@ public static double CompensateY(double y, uint rhall, Bmm150TrimRegisterData tr
             float processCompY2;
             float processCompY3;
             float processCompY4;
-            int BMM150_OVERFLOW_ADCVAL_XYAXES_FLIP = -4096;
+            int bmm150Overflow_adcval_xyaxes_flip = -4096;
```
Comment (josesimoes) [inline devices/Bmm150/Bmm150Compensation.cs:71]: same here

File: devices/Bmm150/Bmm150Compensation.cs [hunk 10]
```diff
@@ -108,20 +108,20 @@ public static double CompensateZ(double z, uint rhall, Bmm150TrimRegisterData tr
             float processCompZ3;
             float processCompZ4;
             float processCompZ5;
-            int BMM150_OVERFLOW_ADCVAL_ZAXIS_HALL = -16384;
+            int bmm150Overflow_adcval_zaxis_flip = -16384;
```
Comment (josesimoes) [inline devices/Bmm150/Bmm150Compensation.cs:111]: here too

File: devices/Bno055/Bno055Sensor.cs [hunk 11]
```diff
@@ -97,17 +131,18 @@ public Units Units
...
-        /// Create an BNO055 sensor
+        /// Initializes a new instance of the <see cref="Bno055Sensor" /> class.
         /// </summary>
-        /// <param name="i2cDevice">The I2C Device</param>
-        /// <param name="operationMode">The operation mode to setup</param>
-        /// <param name="shouldDispose">true to dispose the I2C device at dispose</param>
-        public Bno055Sensor(I2cDevice i2cDevice,
+        /// <param name="i2cDevice">The I2C Device.</param>
+        /// <param name="operationMode">The operation mode to setup.</param>
+        /// <param name="shouldDispose">True to dispose the I2C device at dispose.</param>
```
Comment (josesimoes) [inline devices/Bno055/Bno055Sensor.cs:143]:
```suggestion
        /// <param name="shouldDispose"><see langword="true"/> to dispose the I2C device at dispose.</param>
```

**Review by josesimoes — COMMENTED** [review]
Another batch! Well done!! 👏🏻 🥇

**Comment by torbacz** [comment]
@josesimoes Fixed comments and rebased from develop. Looks like it's ready to be merged.

**Review by josesimoes — APPROVED** [review]
Absolutely!!

**Pull request merged**